pkg/test/framework/label/filter.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package label
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// Selector is a Set of label filter expressions that get applied together to decide whether tests should be selected
+// for execution or not.
+type Selector struct {
+	// The constraints are and'ed together.
+	present Set
+	absent  Set
+}
+
+var _ fmt.Stringer = Selector{}
+
+func NewSelector(present []Instance, absent []Instance) Selector {
+	return Selector{
+		present: append([]Instance{}, present...),
```

Comment:
Why this funny construct? Wouldn't present: present be sufficient?

Reply:
It was to avoid sharing the array with the caller after the call completion.